### ISSUE TITLE
-s not working with -e?

### ISSUE THREAD

**Issue by xfzv** [sharkdp/fd#920]
```zsh
% fd --version
fd 8.3.0
```

I have some `.auCDtect.log` and `.aucdtect.log` files in a directory. When trying to find those with `.aucdtect.log` extension with:

```zsh
fd -s -e ".aucdtect.log"
```

, fd outputs both `.auCDtect.log` and `aucdtect.log` files. It looks like the `-s` option isn't taken into account with `-e`. 

To reproduce:

```zsh
% cd /tmp && touch 1.{aucdtect.log,auCDtect.log}

# Trying to find either "1.aucdtect.log" or "1.auCDtect.log": both are found
% fd -s -e ".aucdtect.log"
1.auCDtect.log
1.aucdtect.log

fd -s -e ".auCDtect.log"
1.auCDtect.log
1.aucdtect.log

# Works as expected without "-e"
% fd -s ".auCDtect.log"
1.auCDtect.log

% fd -s ".aucdtect.log"
1.aucdtect.log
```

Not sure whether I am missing something obvious or if that could be a feature request instead.

Thanks!

**Comment by xfzv** [comment]
Well, I assume this is the intended behavior after finding this: https://github.com/sharkdp/fd/pull/58#issue-261764675

> File extension comparison is always case insensitive.

**Comment by tmccombs** [comment]
That should probably be better documented.

**Comment by nyurik** [comment]
I don't think it was ever explained why `-e` should always ignore case even with `-s`.  I think this is less than ideal, unless i am missing some common use case?

**Comment by tmccombs** [comment]
Because the -e filtering is distinct and separate from the main pattern filtering, and -s applies to the main pattern.

**Comment by nyurik** [comment]
should there be another option like `--exact-extension` or `--extension-exact` ?
